### PR TITLE
add instructions

### PR DESCRIPTION
**Pull request by vikingfacer**
add instructions and mv main in emulator to from source dir.

### CODE REVIEW

File: instructions/include/instr.hpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  /* Load constant immediate*/
+  lci = 0b1010,
+  /* Store constant immediate*/
+  sci = 0b1011,
+  /* Load using offset*/
+  lo = 0b1100,
+  /* Store using offset*/
+  so = 0b1101,
+  /* Jump*/
+  ju = 0b1110,
```
Comment (how2useGit) [inline instructions/include/instr.hpp]: Instruction is jb instead of ju with new mem-mem design

File: instructions/include/instr.hpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  uint16_t reg1 : 4;
+  uint16_t reg2 : 4;
+  uint16_t reg3 : 4;
+};
+
+static_assert(sizeof(instruction) == sizeof(uint16_t),
+              "add instruction must be 16bits");
+
+enum opcode : uint8_t {
+  /* Add reg-mem*/
```
Comment (how2useGit) [inline instructions/include/instr.hpp]: comments are reg-mem still but now using mem-mem

File: instructions/include/instr.hpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+#ifndef INSTR_HPP
+#define INSTR_HPP
+
+// example of possible instruction layout
+#include <cstdint>
+
+#pragma "packed"
+struct instruction {
+  uint16_t opcode : 4;
+  uint16_t reg1 : 4;
```
Comment (how2useGit) [inline instructions/include/instr.hpp]: may want to call our variables mem1,2,3 etc now with mem-mem design

File: instructions/include/instr.hpp (hunk 1 above)
Comment (vikingfacer) [reply instructions/include/instr.hpp]: updated

File: instructions/include/instr.hpp (hunk 3 above)
Comment (vikingfacer) [reply instructions/include/instr.hpp]: updated

File: instructions/include/instr.hpp (hunk 2 above)
Comment (vikingfacer) [reply instructions/include/instr.hpp]: updated

**Pull request merged**